assets/js/googlesitekit/datastore/user/key-metrics.js

```diff
@@ -411,7 +392,46 @@ const baseSelectors = {
 		if ( keyMetricsSettings === undefined ) {
 			return undefined;
 		}
-		return keyMetricsSettings.widgetSlugs;
+
+		if ( ! Array.isArray( keyMetricsSettings.widgetSlugs ) ) {
+			return [];
+		}
+
+		// Even though a user may have picked their own metrics, there is a chance that they no longer
+		// are "available" if they require certain custom dimensions, detected events, feature flags, etc. which no
+		// longer exist. So we should filter these out by using the displayInWidgetArea() callback.
+		const isViewOnly = ! select( CORE_USER ).isAuthenticated();
+		const filteredWidgetSlugs = keyMetricsSettings.widgetSlugs.filter(
+			( slug ) => {
+				const widget = KEY_METRICS_WIDGETS[ slug ];
+
+				if ( ! widget ) {
+					return false;
+				}
+
+				if (
+					widget.displayInWidgetArea &&
+					typeof widget.displayInWidgetArea === 'function'
+				) {
+					return widget.displayInWidgetArea(
```

Comment:
I see this was something we were doing before as well but it seems incorrect to make this selector view-only aware and specific to displaying in the widget area which seem like concerns that are more localized to the React side of things. I guess we can keep it here for simplicity regarding this PR but it does seem like there is room for decoupling these things a bit more.